src/app/components/ThemeProvider/__mocks__/themes.ts

```diff
@@ -160,31 +186,131 @@ type ThemeWithVariant = {
       });
     });
   }
 
-  return config;
-}, {}) as unknown as ThemeWithVariant | ThemeWithNoVariant;
+  return themeConfig;
+}, {}) as ThemeWithVariant | ThemeWithNoVariant;
+
+type ServiceThemeWithNoVariantWithPWATypography = {
+  [_service in ServicesWithNoVariantsWithPWATypography['service']]: Partial<ServiceTheme>;
+};
+
+type ServiceThemeWithVariantWithPWATypography = {
+  [_service in ServicesWithVariantsWithPWATypography['service']]: {
+    [_variant in Variants]?: Partial<ServiceTheme>;
+  };
+};
+
+const pwaThemesNoVariants: ServiceThemeWithNoVariantWithPWATypography = {
+  afaanoromoo: afaanoromooPWA,
+  afrique: afriquePWA,
+  azeri: azeriPWA,
+  gahuza: gahuzaPWA,
+  hausa: hausaPWA,
+  igbo: igboPWA,
+  indonesia: indonesiaPWA,
+  kyrgyz: kyrgyzPWA,
+  pidgin: pidginPWA,
+  somali: somaliPWA,
+  swahili: swahiliPWA,
+  yoruba: yorubaPWA,
+};
+
+const pwaThemesWithVariants: ServiceThemeWithVariantWithPWATypography = {
+  serbian: {
+    cyr: serbianCyrPWA,
+    lat: serbianLatPWA,
+  },
+  ukrainian: { default: ukrainianPWA },
+  uzbek: {
+    cyr: uzbekCyrPWA,
+    lat: uzbekLatPWA,
+  },
+};
+
+type PWAThemeWithNoVariant = {
+  [_service in ServicesWithNoVariantsWithPWATypography['service']]: Theme;
+};
+
+type PWAThemeWithVariant = {
+  [_service in ServicesWithVariantsWithPWATypography['service']]: {
+    [_variant in Variants]?: Theme;
+  };
+};
+
+export const pwaThemes = [
+  ...Object.entries(pwaThemesNoVariants),
+  ...Object.entries(pwaThemesWithVariants),
+].reduce(
+  (pwaThemeConfig, [service, pwaTheme]) => {
+    // @ts-expect-error need to initialise the object in case it does not exist
+    pwaThemeConfig[service] = {};
+
+    if (Object.keys(pwaThemesNoVariants).includes(service)) {
+      const [, baseTheme] = Object.entries(themesNoVariants).find(
+        ([serviceWithNoVariant]) => service === serviceWithNoVariant,
+      ) || ['', { ...ws }];
 
-export default { ...themes };
+      const themeWithPWA = mergeThemeWithPWATypography({
+        baseTheme,
+        pwaTheme: pwaTheme as ServiceTheme,
+      });
+
+      // @ts-expect-error service theme
+      pwaThemeConfig[service] = getThemeConfig({
+        ...themeWithPWA,
+        ...defaultThemeProps,
+      });
+    } else {
+      Object.entries(pwaTheme).forEach(([variant, variantPWATheme]) => {
+        const [, baseThemeWithVariant] = Object.entries(
+          themesWithVariants,
+        ).find(([serviceWithVariant]) => service === serviceWithVariant) || [
+          '',
+          { default: { ...ws } },
+        ];
+
+        const baseTheme = baseThemeWithVariant[
+          variant as Variants
+        ] as ServiceTheme;
+
+        const themeWithPWA = mergeThemeWithPWATypography({
+          baseTheme,
+          pwaTheme: variantPWATheme,
+        });
+
+        // @ts-expect-error service with variant theme
+        pwaThemeConfig[service][variant] = getThemeConfig({
+          ...themeWithPWA,
+          ...defaultThemeProps,
```

Comment:
Are these spread in the right order? Should `...themeWithPWA` come second? The naming `defaultThemeProps` implies they could be overriden